spec/lib/conjur_client_spec.rb

```diff
@@ -125,4 +125,62 @@
       end
     end
   end
+
+  describe "version" do
+
+    context "An empty string is given for the version" do
+      before do
+        allow(ENV).to receive(:[]).and_call_original
+        allow(ENV).to receive(:[]).with('CONJUR_VERSION').and_return("")
+      end
```

Comment:
The before/end pattern is useful when you have multiple tests using the same setup code but in these cases here they're all one-off so you should just put the allows into the test case itself. You _could_ have a `and_call_original` at the toplevel though.

Reply:
Same thing with context - it's not needed if you're not really grouping tests (you're running a single test per context)

Reply:
Ah thank you! I'm new to Ruby so it's very helpful to know how to 'slim down' things, since I only have the other code as a reference point I had a feeling I was missing some things.